.github/workflows/ci.yml

```diff
@@ -233,15 +265,24 @@ jobs:
     - name: Checkout Code
       uses: actions/checkout@v3
 
+    - name: Get CMake and Ninja
+      uses: lukka/get-cmake@latest
+      with:
+        cmakeVersion: ${{ runner.os == 'Windows' && '3.24' || '3.22' }} # Set to minimum required version (3.24 on Windows and 3.22 elsewhere) as specified in CMakeLists.txt to test compatibility
+        ninjaVersion: latest
+
     - name: Install Windows Dependencies
       if: runner.os == 'Windows'
       run: |
-        choco install ninja
         curl.exe -o run-clang-tidy https://raw.githubusercontent.com/llvm/llvm-project/llvmorg-15.0.7/clang-tools-extra/clang-tidy/tool/run-clang-tidy.py
 
     - name: Install Linux Dependencies
       if: runner.os == 'Linux'
-      run: sudo apt-get update && sudo apt-get install libxrandr-dev libxcursor-dev libudev-dev libopenal-dev libflac-dev libvorbis-dev libgl1-mesa-dev libegl1-mesa-dev libdrm-dev libgbm-dev
+      uses: awalsh128/cache-apt-pkgs-action@15d0235a4147963374ec3a9334d73adf0cb9eabc
+      with:
+        packages: libxrandr-dev libxcursor-dev libudev-dev libopenal-dev libflac-dev libvorbis-dev libgl1-mesa-dev libegl1-mesa-dev libdrm-dev libgbm-dev
+        version: 1.0
+        execute_install_scripts: true
```

Comment:
I tested this in isolation and proved that caching apt packages has no effect. Let's not bother doing this and instead keep our pipeline as simple as possible.

Reply:
I'm guessing the packages we install are all pretty small compared to some of the bigger packages out there whose download time would dominate the installation process.

Reply:
Before caching: https://github.com/SFML/SFML/actions/runs/6387165344/job/17335087763

After caching: https://github.com/SFML/SFML/actions/runs/6388201930/job/17337670809

The difference in runtime is not significant. Perhaps it saves a few seconds but it's hard to say. I can cherry pick other examples where it's actually slower than the status quo. The evidence just doesn't prove that caching these packages is meaningful. 

Now how in the first link the packages are pulled from http://azure.archive.ubuntu.com/ubuntu. It appears that GitHub has their own apt mirror which may improve download speeds compared to pulling from Canonical's repositories. Downloading `apt` package is a popular thing to do so this would make sense.